### DISCUSSION TITLE
Mdx creating pages for all directio

### DISCUSSION THREAD

**Discussion by marketingsolution** [gatsbyjs/gatsby#35258] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I have this page structure:
 /blog ->{mdx.slug}.js
 /news ->{mdx.slug}.js
/tests -> {mdx.slug}.js

When I write a post for example: honda-test.mdx in /test directory Gatsby is creating pages in with all slugs so I have:
/blog/honda-test
/news/honda-test
/tests/honda-test

### Reproduction Link

https://github.com/marketingsolution/MotoTripsPL/tree/youtube-video-schema-component

### Steps to Reproduce

1.just add {mdx.slug}.js in few folders in src/pages
2.gatsby develop
3.check if pages are the same for all directories
...

### Expected Result

I just need one page for one directory not for all.

### Actual Result

the same pages are created for all folders I have in src/pages

### Environment

```shell
System:
    OS: Windows 10 10.0.19043
    CPU: (4) x64 Intel(R) Core(TM) i5-6200U CPU @ 2.30GHz
  Binaries:
    Node: 17.0.1 - C:\Program Files\nodejs\node.EXE
    npm: 8.1.0 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.10.0 - /c/Python310/python
  Browsers:
    Chrome: 99.0.4844.82
    Edge: Spartan (44.19041.1266.0), Chromium (99.0.1150.55)
  npmPackages:
    gatsby: ^4.3.0 => 4.3.0
    gatsby-image: ^3.3.0 => 3.11.0
    gatsby-plugin-google-analytics: ^3.14.0 => 3.14.0
    gatsby-plugin-google-tagmanager: ^4.10.2 => 4.10.2
    gatsby-plugin-image: ^1.14.1 => 1.14.2
    gatsby-plugin-manifest: ^3.14.0 => 3.14.0
    gatsby-plugin-mdx: ^3.9.0 => 3.9.0
    gatsby-plugin-offline: ^4.3.0 => 4.14.0
    gatsby-plugin-react-helmet: ^4.3.0 => 4.14.0
    gatsby-plugin-react-leaflet: ^3.0.2 => 3.0.2
    gatsby-plugin-sharp: ^4.3.0 => 4.3.0
    gatsby-plugin-sitemap: ^5.3.0 => 5.3.0
    gatsby-plugin-styled-components: ^4.14.0 => 4.14.0
    gatsby-plugin-web-font-loader: ^1.0.4 => 1.0.4
    gatsby-remark-images: ^6.10.2 => 6.10.2
    gatsby-remark-images-contentful: ^4.11.0 => 4.11.0
    gatsby-source-contentful: ^5.14.1 => 5.14.1
    gatsby-source-filesystem: ^3.14.0 => 3.14.0
    gatsby-source-wordpress: ^5.14.1 => 5.14.2
    gatsby-transformer-remark: ^5.10.2 => 5.10.2
    gatsby-transformer-sharp: ^3.14.0 => 3.14.0
    gatsby-transformer-sqip: ^3.14.3 => 3.14.3
  npmGlobalPackages:
    gatsby-cli: 4.10.2
    gatsby: 3.14.0
```

**Comment by LekoArts** [comment]
Hi!

The File System Route API doesn't automatically "map" those things together. If you use `{mdx.slug}` it will access all MDX files and thus create those at every route you use it. There's currently no way to filter templates like `{mdx.slug}` to a subset of things if you want to keep the structure like `/news/{mdx.slug}`, `/blog/{mdx.slug}` etc.

There are multiple ways how you can make this work though, with varying levels of difficulty.

Here's how you could do it:

![Bildschirmfoto 2022-03-30 um 08 23 33](https://user-images.githubusercontent.com/16143594/160765559-189ddfae-d4dd-469c-b3cc-636466af1123.png)

You give your files a `type` in the frontmatter and then use the path `/{mdx.frontmatter__type}/{mdx.slug}`. This way you create every page only once and the page itself can still use the fast `eq: $id` filtering.

**Reply by marketingsolution** [reply]
Thank you I do something similar but still doesn't work as it should please take a look:
![image](https://user-images.githubusercontent.com/63929301/161031865-87931df1-470e-4106-a1f0-c0885d9d4e48.png)

and one of them doesn't use my layout just plain text, this is how it looks:
![image](https://user-images.githubusercontent.com/63929301/161032194-96197a75-3f30-44c1-ba85-66e3c0e2d9c7.png)

**Reply by marketingsolution** [reply]
hi it works for sinlge pages for posts creation. but some of them doesn't use my layout. can you help me with that?

**Comment by matteobucci** [comment]
Hello, I have the same exact use case. @marketingsolution  have you found a workaround?
